data/transactions/logic/assembler_test.go

```diff
@@ -1067,21 +1104,21 @@ func TestDisassembleTxna(t *testing.T) {
 func TestDisassembleGtxna(t *testing.T) {
 	t.Parallel()
 	// check gtxn and gtxna are properly disassembled
-	gtxnSample := "// version 2\ngtxn 0 Sender\n"
+	gtxnSample := fmt.Sprintf("// version %d\ngtxn 0 Sender\n", AssemblerMaxVersion)
```

Comment:
The changes that you've made to TestDisassembleGtxna aren't wrong. however, once we start having more version of TEAL, we also need to expand these testings. In particular, I think that we have bunch of  TestDisassembleXXX that formats a string to update it's version, assemble, disassemble and compare the disassembled string vs. the constructed one.
I think that this "pattern" can be refactored to as a data-driven test cases, and we should also check it across multiple assembler versions (i.e. 2, 3, etc. ) and not just the latest one.

Reply:
Good idea. Disassembly is functionality I haven't seen in actual use. Are there strong requirements about what it looks like, and being able to map back and forth to disassembled form? Or is it just to display while the debugger walks through your code?

Reply:
I think disassembly should be able to go into assembly and produce the same binary.

Reply:
Then I should change the version comment to #pragma version?

Reply:
Yes! That's something we missed when `#pragma version` was added.